racker/salus-telemetry-resource-management
# converted the resource label matching query into a SQL file in the re…

…sources directory

# Resolves

https://jira.rax.io/browse/SALUS-480

# What

This is meant to simplify reading and debugging of any of the actual SQL queries we need to write for Resource Management

# How

This migrates the StringBuilder approach into just a static .sql file. Most of the query is static with only a few locations needing parameters being set. The static query does utilize a String.Format tag to allow us to inject the complex label matching portion of the query into the middle of this template.

## How to test

Run the unit tests. We weren't testing the query in the first place so the tests dont need to change. We just need to make sure they still return the same.

# Why

I considered removing all of the string building from java and building the whole thing with a jmustache template. Ultimately this is for readability of the query and ease of debugging. I didn't think that jmustache would help with the readability and it actually would add a bit of weirdness because we still need to make sure we have cleansed our inputs by passing everything in through parameters to jdbc.

# TODO

Create the common repo PR
Figure out what sort of Exception we want to through if it can't read the resource file.
